homeassistant/components/sensor/__init__.py

```diff
@@ -246,6 +247,22 @@ def device_class(self) -> SensorDeviceClass | None:
             return self.entity_description.device_class
         return None
 
+    @final
+    @property
+    def numeric_state_expected(self) -> bool:
```

Comment:
Should we make this private since it's unlikely to be useful unlike sensor or classes extending sensor?

```suggestion
    def _numeric_state_expected(self) -> bool:
```

Reply:
I believe the use case is MQTT... so that would not make it ~~private~~ protected :)

Reply:
If it's private, then it can't be used by `mqtt` which is the source of the PR...

Reply:
The use case in MQTT was in a class derived from `SensorEntity`

Reply:
Erik is right, this is local access, so it could just be marked protected (or private).

Reply:
done